lumicks/pylake/kymotracker/tests/test_kymotrackgroup_sources.py

```diff
@@ -255,10 +256,109 @@ def test_multisource_not_implemented(kymos, coordinates):
             tracks, window=2, refine_missing_frames=False, overlap_strategy="ignore"
         )
 
-    with pytest.raises(
-        NotImplementedError,
-        match=(
-            r"Dwelltime analysis is not supported. This group contains tracks from 2 source kymographs."
-        ),
-    ):
-        tracks.fit_binding_times(n_components=1)
+
+def test_tracks_by_kymo(kymos, coordinates):
+    time_indices, position_indices = coordinates
+
+    def make_tracks(kymo):
+        return KymoTrackGroup(
+            [KymoTrack(t, p, kymo, "green") for t, p in zip(time_indices, position_indices)]
+        )
+
+    tracks = {k: make_tracks(k) for k in (kymos[0], kymos[-1])}
+    merged_group = (tracks[kymos[0]] + tracks[kymos[-1]])
+
+    track_dictionary = merged_group._tracks_by_kymo()
+    assert set(track_dictionary.keys()) == {kymos[0], kymos[-1]}
+
+    for key in track_dictionary.keys():
+        tracks_dict = track_dictionary[key]
+        tracks_raw = tracks[key]
+        assert len(tracks_dict) == len(tracks_raw)
+        for track_dict, track_raw in zip(tracks_dict, tracks_raw):
+            id(track_dict) == id(track_raw)
+
+
+def test_multi_kymo_dwell(kymos):
```

Comment:
I think I had kept these kind of tests (and the next one) in `test_kymotrack.py` next to the ones that essentially test the same functionality for single source. I'd honestly rather not keep adding to that file (especially considering the length of the file) but I think it's better to keep tests for the same functionality in the same location rather than split across different files.

Reply:
Yeah, fair enough. I will move them.